include/seqan3/range/view/detail.hpp

```diff
@@ -0,0 +1,366 @@
+        std::tuple<arg_types...> arguments;
+
+        //!\brief Helper function to unpack the tuple.
+        template <typename urng_t, size_t... Is>
+        auto explode(urng_t && urange, std::index_sequence<Is...> const &) const
+        {
+            using parent_type = declare_view_functor_type_with_args<view_type, arg_types...>;
+            return parent_type{}(std::forward<urng_t>(urange), static_cast<arg_types>(std::get<Is>(arguments))...);
+        }
+
+        /*!\brief Define the operator() that is ultimately called inside the pipe-operator to fill in the `urange`
+         * argument
+         * \tparam Type of the underlying range. view_type must be constructible with an argument of urng_t
+         * and arguments of type arg_types.
+         * \param urange The underlying range.
+         */
+        template <typename urng_t>
+        auto operator()(urng_t && urange) const
+            requires (sizeof...(arg_types) > 0) &&
+                     requires { view_type{std::declval<urng_t>(), std::declval<arg_types>()...}; }
+        {
+            return explode(std::forward<urng_t>(urange), std::make_index_sequence<sizeof...(arg_types)>{});
+        }
+    };
+
+    /*!\brief Enables calling the functor without a `urange` parameter, necessary inside a pipe statement.
+     * \tparam arg_types The types of the arguments.
+     * \param args Arguments stored in the auxiliary functor.
+     * \returns An auxiliary functor for use in the pipe notation, **not an instance of `view_type`**!
+     *
+     * \details
+     *
+     * Given:
+     *
+     * ```cpp
+     * template <typename urng_t>
+     *     requires seqan3::input_range_concept<urng_t> // or more/stricter requirements
+     * struct view_foo
+     * {
+     *     // your implementation
+     *     // constructor takes an int argument and e.g. stores it
+     * };
+     *
+     * using foo_fn = declare_view_functor_type_with_args<view_foo>;
+     *
+     * namespace view
+     * {
+     * foo_fn foo;
+     * }
+     * ```
+     *
+     * This functor enables the function call syntax, without the actual underlying range:
+     *
+     * ```cpp
+     * std::vector<int> container{1, 2, 3};
+     *
+     * auto v = view::foo(7);
+     * // v is NOT OF TYPE view_foo<std::vector<int>
+     * // it's usually not used like this, instead use it inside a pipe:
+     * auto v = container | view::foo(7);
+     * ```
+     */
+    template <typename ...arg_types>
+    auxiliary_functor_t<std::add_rvalue_reference_t<arg_types>...> operator()(arg_types && ...args) const
+    {
+        // adding rvalue reference is important, because temporaries+&& make arg_type
+        // be e.g. "int" and not "int &&", but we really want the tuple in the functor
+        // to always only contain lvalue or rvalue references
+        // [adding rvalue reference to lvalue reference is NOP which we want]
+        return {std::add_rvalue_reference_t<arg_types>(args)...};
```

Comment:
I found this last-minute, before it generated dangling stack references if the argument type inside the view was `&`.
:sweat_smile:

Reply:
hm, or maybe I did fix it :thinking: